agent/consul/leader_connect_test.go

```diff
@@ -364,49 +375,45 @@ func TestLeader_SecondaryCA_UpgradeBeforePrimary(t *testing.T) {
 	{
 		state1 := s1.fsm.State()
 		_, roots1, err := state1.CARoots(nil)
-		require.NoError(err)
+		require.NoError(t, err)
 
 		state2 := s2.fsm.State()
 		_, roots2, err := state2.CARoots(nil)
-		require.NoError(err)
-		require.Equal(1, len(roots1))
-		require.Equal(1, len(roots2))
-		require.NotEqual(roots1[0].ID, roots2[0].ID)
-		require.NotEqual(roots1[0].RootCert, roots2[0].RootCert)
+		require.NoError(t, err)
+		require.Equal(t, 1, len(roots1))
+		require.Equal(t, 1, len(roots2))
+		require.NotEqual(t, roots1[0].ID, roots2[0].ID)
+		require.NotEqual(t, roots1[0].RootCert, roots2[0].RootCert)
```

Comment:
This leaves the behavior as it was before. 

It would be pretty easy to switch in the leader code but there are many tests that rely on and are in fact designed to test this behavior. 

As for this test. I just fixed it to handle require + retry better

Reply:
OK I misunderstood. I think that's fine.